spring-kafka/src/test/java/org/hitachivantara/spring/kafka/listener/AbstractRetryMessageListenerTest.java

```diff
@@ -0,0 +1,117 @@
+ * international treaties. Access to the source code contained herein is strictly
+ * prohibited to anyone except those individuals and entities who have executed
+ * confidentiality and non-disclosure agreements or other agreements with Hitachi Vantara,
+ * explicitly covering such access.
+ */
+
+package org.hitachivantara.spring.kafka.listener;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.kafka.listener.AcknowledgingMessageListener;
+import org.springframework.kafka.listener.adapter.RetryingAcknowledgingMessageListenerAdapter;
+import org.springframework.kafka.support.Acknowledgment;
+import org.springframework.retry.RetryPolicy;
+import org.springframework.retry.backoff.BackOffPolicy;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class AbstractRetryMessageListenerTest {
+
+  private AbstractRetryMessageListener abstractRetryMessageListener;
+
+  @Before
+  public void setup() {
+    abstractRetryMessageListener = spy( new AbstractRetryMessageListener() {
+      @Override
+      public void onMessage( Object o, Acknowledgment acknowledgment ) {
+        //no op
+      }
+    } );
+  }
+
+  @Test
+  public void testGetMessageListener() {
+    RetryTemplate retryTemplate = mock( RetryTemplate.class );
+    when( abstractRetryMessageListener.getRetryTemplate() ).thenReturn( retryTemplate );
+
+    assertNotNull( abstractRetryMessageListener.getMessageListener() );
+    assertTrue( abstractRetryMessageListener.getMessageListener() instanceof RetryingAcknowledgingMessageListenerAdapter );
+    RetryingAcknowledgingMessageListenerAdapter acknowledgingMessageListener = (RetryingAcknowledgingMessageListenerAdapter) abstractRetryMessageListener.getMessageListener();
+    assertEquals( retryTemplate, acknowledgingMessageListener.getRetryTemplate() );
+  }
+
+  @Test
+  public void testGetRetryTemplate() {
+    RetryPolicy retryPolicy = abstractRetryMessageListener.getRetryPolicy();
+    assertNotNull( retryPolicy );
+    assertTrue( retryPolicy instanceof SimpleRetryPolicy );
+    assertEquals( SimpleRetryPolicy.DEFAULT_MAX_ATTEMPTS, ( (SimpleRetryPolicy) retryPolicy ).getMaxAttempts() );
+
+    abstractRetryMessageListener.setRetryMaxAttempts( 99 );
+    retryPolicy = abstractRetryMessageListener.getRetryPolicy();
+    assertNotNull( retryPolicy );
+    assertTrue( retryPolicy instanceof SimpleRetryPolicy );
+    assertEquals( 99, ( (SimpleRetryPolicy) retryPolicy ).getMaxAttempts() );
+  }
+
+  @Test
+  public void testGetBackOffPolicy() {
```

Comment:
What is being tested? What are the conditions and expected results?
Overall the same question for the other tests.